subprojects/build-cache-http/src/integTest/groovy/org/gradle/caching/http/internal/HttpBuildCacheServiceIntegrationTest.groovy

```diff
@@ -178,25 +178,25 @@ class HttpBuildCacheServiceIntegrationTest extends AbstractIntegrationSpec imple
     }
 
     def "credentials can be specified via URL"() {
-        httpBuildCache.withBasicAuth("user", "pass")
-        settingsFile.text = useHttpBuildCache(getUrlWithCredentials("user", "pass"))
+        httpBuildCache.withBasicAuth("user", 'pass%:-0]#')
```

Comment:
Should we test for encoding problems in some other place? Also, should we test the username for encoding problems?